internal/job/checkout.go

```diff
@@ -257,43 +312,12 @@ func (e *Executor) CheckoutPhase(ctx context.Context) error {
 		}
 	}
 
-	err = e.sendCommitToBuildkite(ctx)
-	if err != nil {
-		e.shell.OptionalWarningf("git-commit-resolution-failed", "Couldn't send commit information to Buildkite: %v", err)
-	}
-
-	// Store the current value of BUILDKITE_BUILD_CHECKOUT_PATH, so we can detect if
-	// one of the post-checkout hooks changed it.
-	previousCheckoutPath, exists := e.shell.Env.Get("BUILDKITE_BUILD_CHECKOUT_PATH")
-	if !exists {
-		e.shell.Printf("Could not determine previous checkout path from BUILDKITE_BUILD_CHECKOUT_PATH")
-	}
-
-	// Run post-checkout hooks
-	if err := e.executeGlobalHook(ctx, "post-checkout"); err != nil {
-		return err
-	}
-
-	if err := e.executeLocalHook(ctx, "post-checkout"); err != nil {
-		return err
-	}
-
-	if err := e.executePluginHook(ctx, "post-checkout", e.pluginCheckouts); err != nil {
+	// After everything, we need to refresh checkout root.
+	// This is because checkout hook might re-create the checkout root folder entirely, deprecating e.checkoutRoot.
+	if err := e.refreshCheckoutRoot(); err != nil {
```

Comment:
This is pretty much the actual change. 

All other diff are just moving codes around for better readability.